packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -196,6 +196,68 @@ abstract class RenderSliverPersistentHeader extends RenderSliver with RenderObje
     _lastStretchOffset = stretchOffset;
   }
 
+  @override
+  void showOnScreen({
+    RenderObject descendant,
+    Rect rect,
+    Duration duration = Duration.zero,
+    Curve curve = Curves.ease,
+  }) {
+    assert(child != null);
+    final Rect childBounds = descendant != null
+      ? MatrixUtils.transformRect(descendant.getTransformTo(child), rect ?? descendant.paintBounds)
+      : null;
+
+    // Trims the part of `rect` that protrudes `child`'s leading edge.
+    //
+    // If the `rect` a descendant specified in its showOnScreen call exceeds
+    // the leading edge of this sliver (which is usually the same as that of
+    // `child`), the viewport will move towards the leading edge (reduce its
+    // scroll offset) to unpin the persistent header. This is almost always
+    // undesirable.
```

Comment:
It took me awhile to understand this comment, so I tried to clarify a bit. Definitely double check this and make sure I understood correctly, though.

> Trims the given Rect `original` so that it fits within the boundaries given by `top`, `right`, `bottom`, and `left`.
> 
> This is used to prevent the case where `rect` or `descendant` specified in showOnScreen exceed the leading edge of this sliver.  If this were to happen, the viewport would move towards the leading edge (reducing its scroll offset) in order to unpin the persistent header.  This is almost always undesirable.

Reply:
:+1:  I'll put the comment above the `showOnScreen` method instead. 

Is it OK to limit the size of the caret's scroll padding instead, so that it will never exceed the leading edge (see the comment in "editable_text.dart" file)? That makes a little bit more sense to me (in case the caller of `showOnScreen` really expects us to respect the `rect` specified). Or maybe we should reduce the default `scrollPadding` (`EdgeInsets.all(20.0)`) and document this edge case?

Reply:
That makes sense, but would it be a breaking change in normal situations?